vars/transportRequestUploadFile.groovy

```diff
@@ -239,14 +239,20 @@ void call(Map parameters = [:]) {
                         echo "[INFO] Uploading file '${configuration.filePath}' to transport request '${configuration.transportRequestId}'" +
                             " of change document '${configuration.changeDocumentId}'."
 
-                        transportRequestUploadSOLMAN(script: script,
-                            cmClientOpts: configuration.changeManagement.clientOpts,
+                        Map paramsUpload = [
+                            script: script,
+                            cmClientOpts: configuration.changeManagement.clientOpts?: [:],
                             filePath: configuration.filePath,
                             uploadCredentialsId: configuration.changeManagement.credentialsId,
                             endpoint: configuration.changeManagement.endpoint,
                             applicationId: configuration.applicationId,
                             changeDocumentId: configuration.changeDocumentId,
-                            transportRequestId: configuration.transportRequestId)
+                            transportRequestId: configuration.transportRequestId
+                            ]
+
+                        paramsUpload = addDockerParams(script, paramsUpload, configuration.changeManagement.solman?.docker)
```

Comment:
You are sure there is `configuration.changeManagement` because it is in the defaults? But you are not sure there is `solman`?

Reply:
good question, i rely on the code before. There are some mandatory keys like

` .withMandatoryProperty('changeManagement/changeDocumentLabel') `

which makes sure that config.changeManagement exists. I guess in case all the mandatory keys aren't set, an exception might be thrown.